### PR TITLE
Enable MSR for crypto.md

### LINKED ISSUE #1981: MSR fails with crypto tutorial with ini as default storage
## Steps to Reproduce the Problem

Apply the patch:

	diff --git a/doc/tutorials/crypto.md b/doc/tutorials/crypto.md
	index 3881eb8dc..37f09d275 100644
	--- a/doc/tutorials/crypto.md
	+++ b/doc/tutorials/crypto.md
	@@ -35,6 +35,14 @@ If you have no GPG private key available, you can generate one by entering the f
	 
	 The `fcrypt` plugin and the `crypto` plugin support both versions (version 1 and version 2) of GPG.
	 
	+In order to set up our tutorial we import the Elektra test key.
	+We **DO NOT RECOMMEND** to use our key on your local machine, as it is available to the public!
	+
	+```sh
	+gpg2 --import src/plugins/crypto/test_key.asc || gpg --import src/plugins/crypto/test_key.asc
	+echo "trust-model always" > ~/.gnupg/gpg.conf
	+```
	+
	 ## Introduction
	 
	 In this tutorial we explain the use of the `crypto` plugin and the `fcrypt` plugin by a simple example:
	diff --git a/tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt b/tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt
	index 60e8241fe..d991a9b25 100644
	--- a/tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt
	+++ b/tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt
	@@ -12,5 +12,7 @@ add_msr_test (kdb-ls "${CMAKE_SOURCE_DIR}/doc/help/kdb-ls.md")
	 
	 add_msr_test (tutorial_validation "${CMAKE_SOURCE_DIR}/doc/tutorials/validation.md" REQUIRED_PLUGINS validation)
	 
	+add_msr_test (tutorial_crypto "${CMAKE_SOURCE_DIR}/doc/tutorials/crypto.md" REQUIRED_PLUGINS crypto_gcrypt fcrypt)
	+
	 # Only works with super user privileges, since it writes to `/etc/hosts`:
	 # add_msr_test (tutorial_mount "${CMAKE_SOURCE_DIR}/doc/tutorials/mount.md")

Build Elektra with 

    cmake ..
            -DKDB_DB_FILE='default.ini' \
    	-DKDB_DB_INIT='elektra.ini' \
    	-DKDB_DEFAULT_STORAGE=ini

Run the test:

    ctest --output-on-failure -V -R testshell_markdown_tutorial_crypto

## Expected Result

Test should succeed.

## Actual Result

Every execution step in the test fails, including the import of the GPG key.

## System Information

- Elektra Version: master
- local, Docker Container. build server: elektra-ini-mergerequest

## Further Log Files and Output

_n/a_

### PR DESCRIPTION
**Pull request by petermax2**
Close #1981 .

## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [x] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests
- [x] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins and doc/METADATA.md)

## Review

Remove the line below and add the "work in progress" label if you do
not want the PR to be reviewed:

@markus2330 please review my pull request

## Merging

Please add the "ready to merge" label when the build server and you say
that everything is ready to be merged.

### CODE REVIEW

**Comment by petermax2** [comment]
@markus2330 is `src/tools` the right place to put the gpg testkey tool?

**Comment by markus2330** [comment]
Yes, tools is the perfect place.

File: tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt [hunk 1]
```diff
@@ -32,6 +32,7 @@ if (ENABLE_ASAN)
 	message (STATUS "Excluding Markdown Shell Recorder test for `validation`, as it leaks memory and fails with ASAN enabled")
 else (ENABLE_ASAN)
 	add_msr_test (tutorial_validation "${CMAKE_SOURCE_DIR}/doc/tutorials/validation.md" REQUIRED_PLUGINS ni validation)
+	add_msr_test (tutorial_crypto "${CMAKE_SOURCE_DIR}/doc/tutorials/crypto.md" REQUIRED_PLUGINS crypto_gcrypt fcrypt)
```
Comment (markus2330) [inline tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt]: Unfortunately, add_msr_test currently does not support dependencies to tools (which would be required here). It should be fairly easy to add, though.

File: src/tools/elektra-gpg-testkey/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+find_package (LibGpgme QUIET)
+if (NOT LIBGPGME_FOUND)
+    remove_tool (elektra-gpg-testkey "libgpgme not found")
+else ()
+    include_directories (${LIBGPGME_INCLUDE_DIRS})
+    add_executable (elektra-gpg-testkey elektra-gpg-testkey.c)
+    target_link_libraries (elektra-gpg-testkey ${LIBGPGME_LIBRARIES})
+    install (TARGETS elektra-gpg-testkey DESTINATION bin)
```
Comment (markus2330) [inline src/tools/elektra-gpg-testkey/CMakeLists.txt]: What about installing it to TARGET_TOOL_EXEC_FOLDER?

Then you can also call it gen-gpg-testkey and call it via `kdb gen-gpg-testkey`.

Comment (petermax2) [reply src/tools/elektra-gpg-testkey/CMakeLists.txt]: Good idea!

**Comment by petermax2** [comment]
@markus2330 How do I add `gen-gpg-testkey` to `${TOOLS}` per default? The build server does not seem to build it and I can not find the place where "ALL" is being resolved to the tool names.

```sh
STDERR: The command kdb gen-gpg-testkey is not known
```

**Comment by markus2330** [comment]
You need to add it in cmake/ElektraCache.cmake line 81. (TOOLS is not yet as comfortable as with PLUGINS)

And I think you will also run into the dependency problem already mentioned above. On some agents gpgme is not installed and thus the tool cannot build. But because the msr test does not have a dependency to the tool, it would be executed anyway and would fails.

**Comment by petermax2** [comment]
@markus2330 is the build server using different installation directories? It seems like `kdb gen-gpg-testkey` is not executable on the build server altough `gen-gpg-testkey' has been built.

**Comment by markus2330** [comment]
Thank you for the support for having deps on tools.

I see, the problem seems to be that `kdb` does not find its tools as long as it is not installed yet. This can be fixed by setting KDB_EXEC_PATH (ideally for all tests or at least the msr tests).

Seems like your tutorial is the first one that executes a tool via kdb which is not compiled-in kdb. Would be great if you can make this work for both in-source and after installation. This will make the writing of the next tutorial, which needs such a tool, much easier!

**Comment by petermax2** [comment]
Is `tests/shell/include_common.sh.in` the right place to export `KDB_EXEC_PATH` for the tests?

**Comment by markus2330** [comment]
@sanssecours Can you answer this question?

**Comment by sanssecours** [comment]
> Is `tests/shell/include_common.sh.in` the right place to export `KDB_EXEC_PATH` for the tests? 

It should be the right place for all shell tests, as far as I know. At least the Shell Recorder should be able to access the variable `KDB_EXEC_PATH` if you add it there, since it includes `include_common.sh.in`.

**Comment by petermax2** [comment]
At the moment all the Cirrus builds fail. My changes do not seem to be the cause. 

```sh
==> Downloading https://download.java.net/java/GA/jdk12.0.1/69cfe15208a647278a19ef0990eea691/12/GPL/openjdk-12.0.1_osx-x64_bin.tar.gz
curl: (51) SSL: no alternative certificate subject name matches target host name 'download.java.net'
Error: Download failed on Cask 'java' with message: Download failed: https://download.java.net/java/GA/jdk12.0.1/69cfe15208a647278a19ef0990eea691/12/GPL/openjdk-12.0.1_osx-x64_bin.tar.gz
```

**Comment by petermax2** [comment]
We should repeat the Cirrus builds, other than that the MSR tests are working fine now. :tada:

**Pull request merged**